### PR TITLE
Feature/itstyr 62 group control

### PR DESCRIPTION
**Pull request by tuj**
* Removed alterphp extension bundle.
* Upgraded to easyadmin 2.
* Replaces report/system dashboards and lists with manual implementation.
* Implements permission scheme where a user can only edit content that is in a group that they are a part of.
* Added theme and category filters to dashboards.
* Removed themes from systems and report.
* Added systemThemes and reportThemes to groups.
* Added systems to selfservice group if set.
* Commented out line that resulted in broken import.

With these changes the questions that apply to a system or report are decided by which groups they belong to.

### CODE REVIEW

File: src/Controller/AdminController.php [hunk 1]
```diff
@@ -2,472 +2,534 @@
...
-                    'sortField'
-                ),
-                'ASC',
-                $this->entity['search']['dql_filter']
-            );
+        $entityArray = $this->entity;
+        switch ($entityArray['class']) {
+            case Report::class:
+            case System::class:
+                $entity = $this->entityManager->getRepository($entityArray['class'])->find($_GET['id']);
```
Comment (rimi-itk) [inline src/Controller/AdminController.php]: A `getEntity` function may be useful here. 
Apparently, EasyAdminController (also) has an EntityManager instance, cf. https://symfony.com/doc/master/bundles/EasyAdminBundle/book/complex-dynamic-backends.html#admincontroller-properties-and-methods.

File: src/Security/Voter/AnswerVoter.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        $report = $subject->getReport();
+        $system = $subject->getSystem();
+
+        if ($system == null && $report == null) {
+            return false;
+        }
+
+        $entity = $report == null ? $system : $report;
+
+        $userGroups = array_map(function($e) {
```
Comment (rimi-itk) [inline src/Security/Voter/AnswerVoter.php]: `ArrayCollection` has a `map` function: https://www.doctrine-project.org/api/collections/latest/Doctrine/Common/Collections/ArrayCollection.html#method_map

File: src/Service/ReportImporter.php [hunk 3]
```diff
@@ -92,8 +92,8 @@ public function import($src)
                     ['name' => $groupName]
                 );
 
-                if ($findGroup && is_null($report->getGroup())) {
-                    $report->setGroup($findGroup);
+                if ($findGroup && !in_array($findGroup, $report->getGroups()->toArray())) {
```
Comment (rimi-itk) [inline src/Service/ReportImporter.php]: Maybe use https://www.doctrine-project.org/api/collections/latest/Doctrine/Common/Collections/ArrayCollection.html#method_contains?

File: templates/dashboard.html.twig [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+          {% endfor %}
+          {% if not loop.last %}
+            <tr>
+              <td class="dashboard--table-spacing" colspan="{{ (paginator|length) + 2 }}">&nbsp;</td>
+            </tr>
+          {% endif %}
+        {% endfor %}
+      {% else %}
+        <tr>
+          <td colspan="100%">
```
Comment (rimi-itk) [inline templates/dashboard.html.twig]: Maybe I'm getting old, but shouldn't the value of `colspan` be a number (of cells to span)?

File: config/routes/easy_admin.yaml [hunk 5]
```diff
@@ -1,3 +1,4 @@
+# config/routes/easy_admin.yaml
```
Comment (rimi-itk) [inline config/routes/easy_admin.yaml]: Oh, that's the name of this file!

Comment (tuj) [reply config/routes/easy_admin.yaml]: Lol

File: src/Service/ReportImporter.php [hunk 6]
```diff
@@ -50,7 +50,7 @@ public function import($src)
             $report->setSysOwner($this->sanitizeText($entry->{'Systemejerskab'}));
             $report->setSysPurpose($this->sanitizeText($entry->{'Formål'}));
             $report->setSysClassification($this->sanitizeText($entry->{'Systemets klassifikation'}));
-            $report->setSysDateForRevision($this->convertDate($entry->{'Dato for revision'}));
+//@TODO:            $report->setSysDateForRevision($this->convertDate($entry->{'Dato for revision'}));
```
[the remark this answers is not in the document]
Comment (tuj) [reply src/Service/ReportImporter.php]: Yes. There are several new fields added, that needs to be added as well.
The problem with this line arises from a change in field name in anmeldelsesportalen.

File: src/Service/SystemImporter.php [hunk 7]
```diff
@@ -97,11 +97,30 @@ public function import($src)
             if (isset($selfServiceAvailableFromText)) {
                 $selfServiceAvailableFromTitles = explode(';#', $selfServiceAvailableFromText);
 
+                $addToSelfServiceGroup = false;
+
                 foreach ($selfServiceAvailableFromTitles as $title) {
+                    if ($title == '') {
```
[the remark this answers is not in the document]
Comment (tuj) [reply src/Service/SystemImporter.php]: Shouldn't the third parameter be -1 pr the manual?
"A limit of -1 or 0 means "no limit" and, as is standard across PHP, you can use NULL to skip to the flags parameter."

**Pull request merged**